fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -262,11 +272,22 @@ def self.hide_transporter_output?
     #                         see: https://github.com/fastlane/fastlane/pull/4003
     def initialize(user = nil, password = nil, use_shell_script = false)
       use_shell_script ||= !ENV['FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT'].nil?
-      data = CredentialsManager::AccountManager.new(user: user, password: password)
 
+      # First, see if we have an application specific password
+      data = CredentialsManager::AccountManager.new(user: user,
+                                                  prefix: TWO_STEP_HOST_PREFIX)
       @user = data.user
-      @password = data.password
+      @password ||= data.password(ask_if_missing: false)
       @transporter_executor = use_shell_script ? ShellScriptTransporterExecutor.new : JavaTransporterExecutor.new
```

Comment:
Can we move this line to before or after the password handling logic? It feels strange left in the middle here.